features/telemetry/protocols/sflow/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/sflow/adapter/SFlow.java

```diff
@@ -92,7 +93,12 @@ public Long getBytes() {
 
     @Override
     public Direction getDirection() {
-        return Direction.INGRESS;
+        if (!getInt64(document, "source_id").equals(getInt64(document, "input")) &&
+            !getInt64(document, "source_id").equals(Optional.empty())) {
+            return Direction.EGRESS;
+        } else {
+            return Direction.INGRESS;
+        }
```

Comment:
```suggestion
        if (getInt64(document, "source_id").equals(getInt64(document, "input"))) {
            return Direction.INGRESS;
        } else {
            return Direction.EGRESS;
        }
```

Reply:
I did it this way because I only wanted to set egress on packets that specifically met the criteria, and let everything else be ingress (egress as the exception, ingress as the rule, i suppose).  I felt that better aligned with the current behavior of unconditionally marking everything as ingress.  I don't have a problem flipping the logic if that's what's preferred.

Reply:
There are two things that came to my mind:
1) switching the then and the else branches to get rid of some boolean inverts. But your explaination makes sens - if ingress is the default, keep it in the else branch.
2) looks like `source_id` is never empty (if there is a chance, using `getInt64(...).isPresent()` is a more appropriate way to check, btw.)